### PR TITLE
Check for no solution in np.irr Fixes #6744

### LINKED ISSUE #6744: numpy irr should check whether the masked solution is empty
In the current implementation, when no solution is returned by roots, the function return np.nan. But if no real solution is returned, the function fails with an exception:

    ValueError: attempt to get argmin of an empty sequence

The proposed fix is to add the following to https://github.com/numpy/numpy/blob/master/numpy/lib/financial.py#L654

    if res.size == 0 or res[mask].size == 0:

### PR DESCRIPTION
**Pull request by simongibbons**
This implements the proposed fix for #6744

### CODE REVIEW

File: numpy/lib/financial.py [hunk 1]
```diff
@@ -651,7 +651,7 @@ def irr(values):
     """
     res = np.roots(values[::-1])
     mask = (res.imag == 0) & (res.real > 0)
-    if res.size == 0:
+    if res.size == 0 or res[mask].size == 0:
```
Comment (charris) [inline numpy/lib/financial.py]: I think you can do this more simply with
```
if not mask.any():
...
```

Comment (simongibbons) [inline numpy/lib/financial.py]: Do'h yea that makes sense. I've updated the PR with that change.

File: numpy/lib/financial.py [hunk 2]
```diff
@@ -651,7 +651,7 @@ def irr(values):
     """
     res = np.roots(values[::-1])
     mask = (res.imag == 0) & (res.real > 0)
-    if res.size == 0:
+    if res.size == 0 or not mask.any():
```
Comment (charris) [inline numpy/lib/financial.py]: You can also get rid of `res.size == 0`.

Comment (simongibbons) [inline numpy/lib/financial.py]: Yea really shouldn't do this before coffee in the morning!

Sorry for making so many trivial mistakes

**Comment by charris** [comment]
Coffee is the foundation of civilization. A few more hoops to traverse. Could you squash the commits? You can do that with `git rebase -i HEAD^^^` followed by a force push `git push --force origin <your branch>`. Make sure the resulting commit message is as it should be.

**Comment by simongibbons** [comment]
Squashed.

**Comment by charris** [comment]
Thanks. LGTM pending tests.

**Pull request merged**